libraries/rush-lib/src/logic/operations/test/ShellOperationRunnerPlugin.test.ts

```diff
@@ -121,4 +123,76 @@ describe(ShellOperationRunnerPlugin.name, () => {
     // All projects
     expect(Array.from(operations, serializeOperation)).toMatchSnapshot();
   });
+
+  it('parameters should be filtered when parameterNamesToIgnore is specified', async () => {
+    const rushJsonFile: string = path.resolve(__dirname, `../../test/parameterIgnoringRepo/rush.json`);
+    const commandLineJsonFile: string = path.resolve(
+      __dirname,
+      `../../test/parameterIgnoringRepo/common/config/rush/command-line.json`
+    );
+
+    const rushConfiguration = RushConfiguration.loadFromConfigurationFile(rushJsonFile);
+    const commandLineJson: ICommandLineJson = JsonFile.load(commandLineJsonFile);
+
+    const commandLineConfiguration = new CommandLineConfiguration(commandLineJson);
+    const buildCommand: IPhasedCommandConfig = commandLineConfiguration.commands.get(
+      'build'
+    )! as IPhasedCommandConfig;
+
+    // Load project configurations
+    const terminalProvider: ConsoleTerminalProvider = new ConsoleTerminalProvider();
+    const terminal: Terminal = new Terminal(terminalProvider);
+
+    const projectConfigurations = await RushProjectConfiguration.tryLoadForProjectsAsync(
+      rushConfiguration.projects,
+      terminal
+    );
+
+    const fakeCreateOperationsContext: Pick<
+      ICreateOperationsContext,
+      | 'phaseOriginal'
+      | 'phaseSelection'
+      | 'projectSelection'
+      | 'projectsInUnknownState'
+      | 'projectConfigurations'
+      | 'rushConfiguration'
+      | 'customParameters'
+    > = {
+      phaseOriginal: buildCommand.phases,
+      phaseSelection: buildCommand.phases,
+      projectSelection: new Set(rushConfiguration.projects),
+      projectsInUnknownState: new Set(rushConfiguration.projects),
+      projectConfigurations,
+      rushConfiguration,
+      customParameters: new Map()
```

Comment:
Enumerate the parameter declarations in `buildCommand.associatedParameters` and use them to define the relevant CommandLineParameter subclass instances. Factor out the logic here:
https://github.com/microsoft/rushstack/blob/3f7b8f94293832edeb949334cffc9379d499b80c/libraries/rush-lib/src/cli/scriptActions/BaseScriptAction.ts#L45C5-L121C68
into a helper function that takes in a `CommandLineAction` (what is currently `this` in the linked code) and the set of `associatedParameters` and returns the `customParameters` map.

Then for the test, use a dummy `CommandLineAction` as the target and manually invoke the public `_setValue` method on each parameter to provide a value.